src/meshes/t8code_mesh.jl

```diff
@@ -234,34 +305,62 @@ function T8codeMesh{NDIMS}(cmesh::Ptr{t8_cmesh};
 
     nodes_in = [-1.0, 1.0]
     matrix = polynomial_interpolation_matrix(nodes_in, nodes)
-    data_in = Array{RealT, 3}(undef, 2, 2, 2)
-    tmp1 = zeros(RealT, 2, length(nodes), length(nodes_in))
 
-    for itree in 0:(num_local_trees - 1)
-        veptr = t8_cmesh_get_tree_vertices(cmesh, itree)
-        verts = unsafe_wrap(Array, veptr, (3, 1 << NDIMS))
+    if NDIMS == 2
+        data_in = Array{RealT, 3}(undef, 2, 2, 2)
+        tmp1 = zeros(RealT, 2, length(nodes), length(nodes_in))
 
-        u = verts[:, 2] - verts[:, 1]
-        v = verts[:, 3] - verts[:, 1]
-        w = [0.0, 0.0, 1.0]
+        for itree in 0:(num_local_trees - 1)
+            veptr = t8_cmesh_get_tree_vertices(cmesh, itree)
+            verts = unsafe_wrap(Array, veptr, (3, 1 << NDIMS))
 
-        vol = dot(cross(u, v), w)
+            u = verts[:, 2] - verts[:, 1]
+            v = verts[:, 3] - verts[:, 1]
+            w = [0.0, 0.0, 1.0]
 
-        if vol < 0.0
-            @warn "Discovered negative volumes in `cmesh`: vol = $vol"
+            # Triple product gives signed volume of spanned parallelepiped.
+            if dot(cross(u, v), w) < 0.0
```

Comment:
I guess it could be more type-stable to compare this to `zero(eltype(w))` or something.
Maybe @ranocha knows this?

Reply:
I refactored this code snippet. What do you think?

Reply:
Doesn't the last component of `w` has to be a 1?
In that case I would probably use `w = [zero(eltype(v)), zero(eltype(v)), one(eltype(v))]` and compare to `zero(eltype(v))` although this is probably over-engineered given that we call this only once.

Reply:
This just came to my mind when scrolling through the changes, but is most likely not important.

Reply:
Stupid me! Of course. That slipped in as a typo ...
Just fixed that. Thx for seeing this.

Reply:
> Doesn't the last component of `w` has to be a 1? In that case I would probably use `w = [zero(eltype(v)), zero(eltype(v)), one(eltype(v))]` and compare to `zero(eltype(v))` although this is probably over-engineered given that we call this only once.

It is ok. I prefer to write Julia like g'old Fortran by writing out the data types anyway. :D

Reply:
> I prefer to write Julia like g'old Fortran by writing out the data types anyway. 

Ok, let's see what the other reviewers think about this.